### ISSUE TITLE
Bug: Empty svg added to the editor on discard Excalidraw

### ISSUE THREAD

**Issue by pabloasc** [facebook/lexical#2089]
Empty svg image is added to the editor on discard Excalidraw

![image](https://user-images.githubusercontent.com/3740777/167025199-1a4d8c6a-2116-40e0-8e61-276095a52b2e.png)

Lexical version: 0.2.5

## Steps To Reproduce

1.  Click on add -> Excalidraw
2. Add a shape and then discard the changes
3. When you are back to the editor, an empty svg with the editor-image is being added

## The current behavior
Empty svg added to the editor
## The expected behavior
No empty svg added to the editor

**Comment by pabloasc** [comment]
I self-assign this issue to me. Working on a fix

**Comment by trueadm** [comment]
Thanks for the fix. I think the fundamental problem here is that we're inserting a node before we open the dialog. This seems wrong.

I think the better flow would be to insert the node after first actually creating something from the dialog. This would also fix the collaboration problem where if one user inserts an Excalidraw node, it doesn't open the dialog for all other users. @pabloasc fancy taking a stab at that? :D

**Comment by pabloasc** [comment]
makes sense @trueadm , check my latest PR. Not totally sure if this fixes the collaboration issue, can you confirm me if it does?
